packages/design-system/src/components/IdleTimeout/IdleTimeout.tsx

```diff
@@ -77,70 +79,146 @@ const defaultMessageFormatter = (timeTilTimeout: number): React.ReactNode => {
   );
 };
 
+// names for local storage variables
+const timeoutCookieName = 'CMS_DS_TIMEOUT';
+const timeoutWarningCookieName = 'CMS_DS_WARNING';
+
 const IdleTimeout = ({
-  closeDialogText = 'Close',
   continueSessionText = 'Continue session',
   heading = 'Are you still there?',
   endSessionButtonText = 'Logout',
+  endSessionUrl = '/logout',
   formatMessage = defaultMessageFormatter,
-  onClose,
   onSessionContinue,
   onSessionForcedEnd,
   onTimeout,
   showSessionEndButton = false,
   timeToTimeout,
   timeToWarning = 5,
-}: IdleTimeoutProps) => {
-  const handleContinueSession = () => {
-    // reset countdown timer
-    if (onSessionContinue) {
-      onSessionContinue();
+}: IdleTimeoutProps): JSX.Element => {
+  if (timeToWarning > timeToTimeout) {
+    console.error(
+      'Error in TimeoutManager component. `timeToWarning` is greater or equal to `timeToTimeout`'
+    );
+  }
+  const MS_BETWEEN_STATUS_CHECKS = 30000;
+  // convert minutes to milliseconds
+  const msToTimeout = timeToTimeout * 60000;
+  const msToWarning = (timeToTimeout - timeToWarning) * 60000;
+  const [checkStatusTime, setCheckStatusTime] = useState<number>(null);
+  const [showWarning, setShowWarning] = useState<boolean>(false);
+  const [timeInWarning, setTimeInWarning] = useState<number>(timeToWarning);
+
+  // cleanup timeouts & intervals
+  const clearTimeouts = () => {
+    setCheckStatusTime(null);
+  };
+
+  // when the countdown for the session ends, clean up, call callback & close modal
+  const handleTimeout = () => {
+    clearTimeouts();
+    removeEventListeners();
+    onTimeout();
+    setShowWarning(false);
+  };
+
+  // when it's time to warn the user about idleness,
+  // set an interval that updates the modal message
+  const handleWarningTimeout = () => {
+    setShowWarning(true);
+    removeEventListeners();
+  };
+
+  const setTimeoutCookies = () => {
+    const timeoutTime = Date.now() + msToTimeout;
+    const warningTime = Date.now() + msToWarning;
+    localStorage.setItem(timeoutCookieName, timeoutTime.toString());
+    localStorage.setItem(timeoutWarningCookieName, warningTime.toString());
```

Comment:
If we start saving the _last active_ timestamp instead of the timeout and warning time, I think we'd only need to store and update one value. We can then calculate the difference between now and that last active time on intervals and see if it's greater than the warning time or greater than the timeout time.